internal/server/server.go

```diff
@@ -19,26 +20,43 @@ var (
 
 // Server contains all the necessary information to run Bifrost
 type Server struct {
+	cfg *Config
```

Comment:
Let's also keep only a value here. Basically, I want the config to be initialized once and then read-only.